Common/src/Storage/ITaskTable.cs

```diff
@@ -35,74 +35,248 @@
 
+  /// <summary>
+  ///  Update a task status to TaskStatus.Processing
+  /// </summary>
+  /// <param name="taskId">Id of the task to start</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// Task representing the asynchronous execution of the method
+  /// </returns>
   Task StartTask(string            taskId,
                  CancellationToken cancellationToken = default);
 
+  /// <summary>
+  ///  Cancel all tasks in a given session
+  /// </summary>
+  /// <param name="sessionId">Id of the target session</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// Task representing the asynchronous execution of the method
+  /// </returns>
   Task CancelSessionAsync(string            sessionId,
                           CancellationToken cancellationToken = default);
 
+  /// <summary>
+  /// Count tasks matching a given filter
+  /// </summary>
+  /// <param name="filter">Task Filter describing the tasks to be counted</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// The number of tasks that matched the filter
+  /// </returns>
   Task<IEnumerable<TaskStatusCount>> CountTasksAsync(TaskFilter        filter,
                                                      CancellationToken cancellationToken = default);
-
+  /// <summary>
+  /// Count tasks matching a given status
+  /// </summary>
+  /// <param name="status">Status of the tasks to be counted</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// The number of tasks that matched the status
+  /// </returns>
   Task<int> CountAllTasksAsync(TaskStatus        status,
                                CancellationToken cancellationToken = default);
 
+  /// <summary>
+  /// Remove a task from the data base given its id
+  /// </summary>
+  /// <param name="id">Id of the tasks to be deleted</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// Task representing the asynchronous execution of the method
+  /// </returns>
   Task DeleteTaskAsync(string            id,
                        CancellationToken cancellationToken = default);
 
+  /// <summary>
+  /// List all tasks matching a given filter
+  /// </summary>
+  /// <param name="filter">Task Filter describing the tasks to be counted</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  /// List of tasks that matched the filter
+  /// </returns>
   IAsyncEnumerable<string> ListTasksAsync(TaskFilter        filter,
                                           CancellationToken cancellationToken);
 
+  /// <summary>
+  ///  Tag a task as succeded
```

Comment:
Change the status of the task to succeeded

Reply:
done